src/Illuminate/Events/EventHooks.php

```
@@ -0,0 +1,627 @@
+    }
+
+    /**
+     * Determine if the given event/hook has callbacks registered for its parent(s)/interface(s).
+     *
+     * @param  array<int, mixed>  $payload
+     */
+    protected function hasHierarchicalCallbacks(string $hook, string $event, array $payload): bool
+    {
+        $this->validateHook($hook);
+
+        if (! class_exists($event) && ! is_object(Arr::first($payload))) {
```

PHPMD (2.15.0)

Avoid using static access to class '\Illuminate\Support\Arr' in method 'hasHierarchicalCallbacks'. (Clean Code Rules)

(StaticAccess)